react-accordion-component/src/accordian.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+
+export default function Accordian({ items }) {
+  const [isOpen, setIsOpen] = useState('');
+
+  return (
+    <div className='row'>
+      <ListItems items={items} isOpen={isOpen} setIsOpen={setIsOpen} />
+    </div>
+  );
+}
+
+function randomColor() {
+  const classes = {
+    width: '25%',
+    backgroundColor: 'rgb(' + Math.floor(Math.random() * 255) +
+      ',' + Math.floor(Math.random() * 255) + ',' +
+      Math.floor(Math.random() * 255) + ')'
+  };
+  return classes;
+}
+
+function ListItems({ items, isOpen, onClick, setIsOpen }) {
+  return (
+    <>
+      {items.map((item) => (
+        <div key={item.language} style={randomColor()}>
+          <h1 onClick={() => isOpen === item.language ? setIsOpen('') : setIsOpen(item.language)}>{item.language}</h1>
```

Comment:
This isn't a bad idea for communicating with the parent. My only concern with it is that the logic of how `isOpen` is interpreted is now in two components -- Accordian and ListItems. If you change your mind on how you want it to work, you will have to find and modify both places.